File: apigateway_getting_started/http_api_lambda_auth.py
# snippet-start:[serverless-snippets.python.http_api_lambda_auth_simple]
def lambda_handler(event, _context):
    try:
        if event["headers"]["authorization"] == "secretToken":
            print("allowed")
            return {
                "isAuthorized": True,
                "context": {
                    "stringKey": "value",
                    "numberKey": 1,
                    "booleanKey": True,
                    "arrayKey": ["value1", "value2"],
                    "mapKey": {"value1": "value2"},
                },
            }
        else:
            print("denied")
            return {
                "isAuthorized": False,
                "context": {
                    "stringKey": "value",
                    "numberKey": 1,
                    "booleanKey": True,
                    "arrayKey": ["value1", "value2"],
                    "mapKey": {"value1": "value2"},
                },
            }
    except BaseException:
        print("denied")
        raise
# snippet-end:[serverless-snippets.python.http_api_lambda_auth_simple]

File: apigateway_getting_started/http_api_lambda_auth.js
// snippet-start:[serverless-snippets.javascript.http_api_lambda_auth_simple]
export const handler = async (event) => {
    if (event.headers.authorization === "secretToken") {
        console.log("allowed");
        return {
            "isAuthorized": true,
            "context": {
                "stringKey": "value",
                "numberKey": 1,
                "booleanKey": true,
                "arrayKey": ["value1", "value2"],
                "mapKey": { "value1": "value2" }
            }
        };
    }
    else {
        console.log("denied");
        return {
        "isAuthorized": false,
        "context": {
            "stringKey": "value",
            "numberKey": 1,
            "booleanKey": true,
            "arrayKey": ["value1", "value2"],
            "mapKey": { "value1": "value2" }
        }
    };
    }
};
// snippet-end:[serverless-snippets.javascript.http_api_lambda_auth_simple]